core/arch/x86/decode_fast.c

```diff
@@ -537,10 +538,17 @@ decode_sizeof(dcontext_t *dcontext, byte *start_pc,
                 sz += 1;
                 /* up to caller to check for addr prefix! */
                 break;
+            case EVEX_PREFIX_OPCODE: {
+                /* If 64-bit mode or EVEX.R' bit is flipped, this is evex */
+                if (X64_MODE_DC(dcontext) || TEST(0x10, *(pc + 1))) {
+                    evex_prefix = true;
+                }
+            }
```

Comment:
Add a comment that this fall-through is deliberate (looks like a bug o/w)

Reply:
Done